src/devices/bus/heathzenith/h8/h8bus.cpp

```diff
@@ -0,0 +1,242 @@
+  corporate headquarters were located.
+
+***************************************************************************/
+
+#include "emu.h"
+
+#include "h8bus.h"
+
+device_h8bus_card_interface::device_h8bus_card_interface(const machine_config &mconfig, device_t &device) :
+	device_interface(device, "h8bus"),
+	m_h8bus(nullptr)
+{
+}
+
+device_h8bus_card_interface::~device_h8bus_card_interface()
+{
+}
+
+void device_h8bus_card_interface::interface_pre_start()
+{
+	if (!m_h8bus)
+	{
+		fatalerror("Can't find H-8 bus device\n");
+	}
+}
+
+device_h8bus_p1_card_interface::device_h8bus_p1_card_interface(const machine_config &mconfig, device_t &device):
+	device_h8bus_card_interface(mconfig, device)
+{
+}
+
+device_h8bus_p1_card_interface::~device_h8bus_p1_card_interface()
+{
+}
+
+device_h8bus_p2_card_interface::device_h8bus_p2_card_interface(const machine_config &mconfig, device_t &device) :
+	device_h8bus_card_interface(mconfig, device)
+{
+}
+
+device_h8bus_p2_card_interface::~device_h8bus_p2_card_interface()
+{
+}
+
+DEFINE_DEVICE_TYPE(H8BUS_SLOT, h8bus_slot_device, "h8bus_slot", "Heath H-8 slot")
+
+h8bus_slot_device::h8bus_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	h8bus_slot_device(mconfig, H8BUS_SLOT, tag, owner, clock)
+{
+}
+
+h8bus_slot_device::h8bus_slot_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_single_card_slot_interface(mconfig, *this),
+	m_h8bus(*this, finder_base::DUMMY_TAG),
+	m_h8bus_slottag(nullptr)
+{
+}
+
+void h8bus_slot_device::device_start()
+{
+}
+
+void h8bus_slot_device::device_resolve_objects()
+{
+	device_h8bus_card_interface *dev = get_card_device();
+
+	if (dev)
+	{
+		dev->set_h8bus_tag(m_h8bus.target(), m_h8bus_slottag);
+		m_h8bus->add_h8bus_card(*dev);
+	}
+}
+
+
+DEFINE_DEVICE_TYPE(H8BUS, h8bus_device, "h8bus", "Heathkit H8 bus (Benton Harbor Bus)")
+
+h8bus_device::h8bus_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	h8bus_device(mconfig, H8BUS, tag, owner, clock)
+{
+}
+
+h8bus_device::h8bus_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_memory_interface(mconfig, *this),
+	m_mem_config("mem", ENDIANNESS_LITTLE, 16, 16, 0, address_map_constructor(FUNC(h8bus_device::mem_map), this)),
+	m_io_config("pio", ENDIANNESS_LITTLE, 8, 8, 0, address_map_constructor(FUNC(h8bus_device::io_map), this)),
+	m_p1_card(nullptr),
+	m_p2_card(nullptr)
+{
+}
+
+h8bus_device::~h8bus_device()
+{
+}
+
+void h8bus_device::interface_pre_start()
+{
+	// Verify required cards are installed.
+	if (!m_p1_card)
+	{
+		fatalerror("Can't find H-8 P1 - Front panel card\n");
+	}
+
+	if (!m_p2_card)
+	{
+		fatalerror("Can't find H-8 P2 - CPU card\n");
+	}
```

Comment:
I don’t think this is a great idea – I realise the system won’t do much without these cards, but MAME doesn’t have a nice way of displaying fatal errors when starting a system from the internal UI, or reporting them back to an external launcher.  Letting the user discover the system does nothing is the lesser evil IMO (much like removing the video card from a Mac II or IBM PC clone in MAME).

Reply:
Ok, I'll remove these. Should I also remove the fatalerror() that is in `device_h8bus_card_interface::interface_pre_start()` if m_h8bus is not set? Or should it be a `throw device_missing_dependencies();` like it is in the intellec4.cpp code?